pkg/http/http.go

```diff
@@ -21,27 +21,20 @@ package http
 import (
 	"fmt"
 	"net/http"
-	"strings"
+	"sync"
 	"time"
 
 	"github.com/prometheus/client_golang/prometheus"
 )
 
 var (
-	requestDuration = prometheus.NewSummaryVec(
```

Comment:
Ideally this should stay. The plan is to wrap it like here https://github.com/kubernetes-sigs/external-dns/blob/789494f10bcb6dc429f33ab1a17de7f082ecd1d8/pkg/metrics/metrics.go#L38 or here https://github.com/kubernetes-sigs/external-dns/blob/789494f10bcb6dc429f33ab1a17de7f082ecd1d8/controller/controller.go#L38 as we have documentation for metrics automated https://github.com/kubernetes-sigs/external-dns/blob/master/docs/monitoring/metrics.md

Reply:
Changing this, may affect kube monitoring as well, which is not desirable.

Reply:
If you have time, would be better to slice PRs.

Reply:
Okay I've modified this PR so it has what I hope is the minimum set of changes you were hoping for. I'll have another PR out either later tonight / tomorrow which will be branched off this with the other changes I made.